eva_submission/evapro/populate_evapro.py

```diff
@@ -162,6 +186,27 @@ def load_samples_from_vcf_file(self, sample_name_2_sample_accession, vcf_file, v
                                        name_in_file=sample_name)
         self.eva_session.commit()
 
+    def load_samples_from_analysis(self, sample_name_2_sample_accession, analysis_accession):
+        # For analyses with aggregated VCFs, get sample accessions associated with the analysis
+        sample_accessions = [sample_info[1] for sample_info in
+                             self.ena_project_finder.find_samples_in_ena(analysis_accession)]
+        self.info(f'Sample accessions for {analysis_accession} found in ENA: {sample_accessions}')
+        sample_accession_2_sample_name = dict(
+            zip(sample_name_2_sample_accession.values(), sample_name_2_sample_accession.keys()))
+        self.eva_session.begin()
+        file_objs = self.get_files_for_analysis(analysis_accession)
+        for sample_accession in sample_accessions:
+            sample_name = sample_accession_2_sample_name.get(sample_accession)
+            sample_obj = self.get_sample(sample_accession)
+            if not sample_obj:
+                self.error(f'Cannot find sample {sample_accession} in EVAPRO')
+                return
```

Comment:
Should we have an explicit rollback here ?

Reply:
Added, I don't think it's necessary but it doesn't hurt